ui/test/nuxeo-delete-document-button.test.js

```diff
@@ -0,0 +1,215 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { fixture, flush, html, isElementVisible, waitForEvent } from '@nuxeo/nuxeo-elements/test/test-helpers.js';
+import { tap } from '@polymer/iron-test-helpers/mock-interactions.js';
+import { dom } from '@polymer/polymer/lib/legacy/polymer.dom';
+import '../actions/nuxeo-delete-document-button.js';
+/* eslint-disable no-unused-expressions */
+
+suite('<nuxeo-delete-document-button>', () => {
+  let button;
+
+  const deletableDocument = {
+    'entity-type': 'document',
+    uid: '1',
+    contextParameters: {
+      permissions: ['Remove'],
+    },
+    facets: [],
+    isTrashed: false,
+    isVersion: false,
+    type: 'File',
+  };
+
+  setup(async () => {
+    button = await fixture(
+      html`
+        <nuxeo-delete-document-button></nuxeo-delete-document-button>
+      `,
+    );
+  });
+
+  suite('Button Visibility', () => {
+    test('Should not be visible when document is null', async () => {
+      button.document = null;
+      await flush();
+      expect(isElementVisible(dom(button.root).querySelector('.action'))).to.be.false;
```

Comment:
Perhaps extract `dom(button.root).querySelector('.action')` to a method, since this is repeated a bunch of times.

Reply:
Yup, we can extract it 👍